abci/client/grpc_client.go

```diff
@@ -245,3 +245,7 @@ func (cli *grpcClient) VerifyVoteExtension(ctx context.Context, req *types.Reque
 func (cli *grpcClient) FinalizeBlock(ctx context.Context, req *types.RequestFinalizeBlock) (*types.ResponseFinalizeBlock, error) {
 	return cli.client.FinalizeBlock(ctx, types.ToRequestFinalizeBlock(req).GetFinalizeBlock(), grpc.WaitForReady(true))
 }
+
+func (cli *grpcClient) SignGossipVote(ctx context.Context, req *types.RequestSignGossipVote) (*types.ResponseSignGossipVote, error) {
```

Comment:
i think signing and if possible verifying inject txn votes should be done at the tendermint level? when passed to the oracle keeper it shld be valid votes already

this is seems better used as a "PrepareOracleVotes" hook? 

flow: app collects new fetched data from oracle service and buffers them simply. every gossip interval, TM calls app with PrepareOracleVotes and app responds with the new votes in ResponsePrepareOracleVote, and htne app clears its buffer. new votes are pushed into gossipvote buffer and signed using val key, and then gossiped immediately.